### ISSUE TITLE
Dependencies: umap-learn incompatible with Python 3.13, using fallback StandardScaler

### ISSUE THREAD

**Issue by rogermt** [rogermt/forgesyte-plugins#11]
## Problem
- `umap-learn` version 0.5.x requires Python <3.10
- Project uses Python 3.13.11
- Current workaround: Fallback to `sklearn.preprocessing.StandardScaler`

## Impact
**Functionality degraded:**
- UMAP: Non-linear dimensionality reduction (3D projection)
- StandardScaler: Linear scaling (different math, different results)
- Team classification accuracy may be affected

## Code Context
File: `src/forgesyte_yolo_tracker/utils/team.py` (lines 63-70)
```python
if umap is not None:
    self.reducer = umap.UMAP(n_components=3)
else:
    from sklearn.preprocessing import StandardScaler
    self.reducer = StandardScaler()  # FALLBACK - different behavior
```

## Solution Options
1. **Wait for umap-learn 0.6+** (supports Python 3.13) - Timeline unknown
2. **Pin Python to <3.10** for this plugin only
3. **Replace UMAP entirely** with another dimensionality reduction (PCA, t-SNE)
4. **Document current behavior** that we're using StandardScaler, not UMAP

## Current Status
✅ **Works** but with degraded performance. Not a blocker, but should be tracked.

## Related Files
- `src/forgesyte_yolo_tracker/utils/team.py`
- `pyproject.toml` (dependencies)
